pkg/entity/run/run.go

```diff
@@ -71,6 +74,32 @@ func (e *DefExecuteContext) Trace(msg string, opt ...TraceOp) {
 	e.trace(msg, opt...)
 }
 
+// Tracef
+func (e *DefExecuteContext) Tracef(msg string, a ...interface{}) {
+	args, ops := splitArgsAndOpt(a...)
+	e.trace(fmt.Sprintf(msg, args...), ops...)
+}
+
+func splitArgsAndOpt(a ...interface{}) ([]interface{}, []TraceOp) {
```

Comment:
This implement limit user must write trace's options at end,  we should highlight it in method's description.
Or we also could implement `Tracef` as beblow:
```golang

func splitArgsAndOpt(a ...interface{}) (args []interface{},opts []TraceOp)
for _, item := range a {
    if _, ok := item.(TraceOp); ok {
      opts = append(opts, item)
      continue
   }
   args = append(args, item)
}
return
}
```

so developer can ignore the order of parameter. how do you think?

Reply:
It is customary to put opt ​​at the end. In addition, it can have better performance when placed at the end.